packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -113,13 +115,26 @@ function validateLightDomTemplate(template: Template, vm: VM) {
     }
 }
 
+const browserExpressionSerializer = (partToken: string, classAttrToken: string) => {
+    // This will insert the scoped style token as a static class attribute in the fragment
+    // bypassing the need to call applyStyleScoping when mounting static parts.
+    return StringCharAt.call(partToken, 0) === STATIC_PART_TOKEN_ID.CLASS ? classAttrToken : '';
+};
+const serializerNoop = () => '';
 // This function serializes the expressions generated by static content optimization.
 // Currently this is only needed for SSR.
 // TODO [#4078]: Split the implementation between @lwc/engine-dom and @lwc/engine-server
 function buildSerializeExpressionFn(parts?: VStaticPart[]) {
-    if (process.env.IS_BROWSER || isUndefined(parts)) {
-        return () => '';
+    if (process.env.IS_BROWSER) {
+        return browserExpressionSerializer;
     }
+
+    if (isUndefined(parts)) {
+        // Technically this should not be reachable, if there are no parts there should be no partTokens
+        // and this function should never be invoked.
```

Comment:
If this function should never be reachable then should we throw an Error instead of returning an empty string?

Reply:
Good point!

Reply:
Now throwing an error - https://github.com/salesforce/lwc/pull/4071/commits/51ef025cc8397ac0af8c6287df8ea5bb4a50e680#diff-9d36a520650dfa1dd13874ab3870b6f688303b9e5860066178098aadcf45a860R124